eks/terraform/modules/cluster/main.tf

```diff
@@ -725,9 +654,9 @@ locals {
     rbac : {
       serviceAccount : {
         name : local.cluster_autoscaler_service_account,
-        annotations : {
-          "eks.amazonaws.com/role-arn" : try(module.cluster_autoscaler_irsa_role.iam_role_arn, "")
-        }
+        annotations : try(var.use_irsa_v1 ? {
+          "eks.amazonaws.com/role-arn" : module.cluster_autoscaler_irsa_role[0].iam_role_arn
+        } : null)
```

Comment:
@Luay-Sol I am not sure about the syntax but AFAIK the `try` function takes 2 arguments. I can see that you have made the first one but which one is the fallback expression? Am I missing anything here?

Reply:
not real, here's the definition of `try`:
```
try evaluates all of its argument expressions in turn and returns the result of the first one that does not produce any errors.
```
And below an example:

<img width="184" alt="Screenshot 2024-07-18 at 9 16 10 PM" src="https://github.com/user-attachments/assets/a6734a78-2e66-4013-a4bc-c937b7edb84d">

Reply:
So one expression with a result will work

Reply:
I see. However, if there is only one expression (so there is no fallback), can we not just use the expression?